### PR TITLE
Add API coverage overview, Light and LightGroup attributes mutations

### PR DESCRIPTION
**Pull request by mickdekkers**
I haven't used jest and nock much yet, so please let me know if there's anything I can do to improve the tests in particular 🙂

### CODE REVIEW

File: src/resolvers/__tests__/set-light-attributes.test.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  afterEach(() => {
+    get.done();
+    put.done();
+  });
+
+  it('sets the light attributes', async () => {
+    await query`mutation {
+      hue {
+        setLightAttributes(id: 10 attributes: { name: "foo" }) { name }
```
Comment (PsychoLlama) [inline src/resolvers/__tests__/set-light-attributes.test.js]: I've seen a pattern in other GraphQL servers which is growing on me. Mutations follow the naming convention `update<Type>`/`create<Type>`/`delete<Type>`. Queries are just the lowercased type name, like `light(...)`, while mutations would be `createLight`, `updateLight`, and `deleteLight`. Keeps the API consistent and easy to guess at.

Not asking for any changes, just thinking out loud :wink:

File: src/utils/__tests__/group-attributes-mutation.test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    expect(result).toEqual(data);
+  });
+
+  it('throws if the lights list contains empty strings', () => {
+    const fail = () => mutation({ lights: ['1', '2', ''] });
+
+    expect(fail).toThrow(/lights/i);
+  });
+});
```
Comment (PsychoLlama) [inline src/utils/__tests__/group-attributes-mutation.test.js]: These are some great tests, I love the validations you've got

File: src/utils/group-attributes-mutation.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  if (typeof name === 'string') {
+    assert(name.length <= 32, 'Name must be 32 characters or less');
+
+    Object.assign(patch, { name });
+  }
+
+  if (typeof _class === 'string') {
+    // Convert enum value to the case the API requires
+    const casedClass = sentenceCase(_class);
+    assert(isValidRoom(casedClass), 'Class must be a valid room');
```
Comment (PsychoLlama) [inline src/utils/group-attributes-mutation.js]: Neat trick!

File: src/resolvers/__tests__/set-light-attributes.test.js (hunk 1 above)
Comment (mickdekkers) [reply src/resolvers/__tests__/set-light-attributes.test.js]: Yeah, that makes sense. I read [this article](https://dev-blog.apollodata.com/designing-graphql-mutations-e09de826ed97) recently which may also be relevant 🙂

File: src/utils/light-attributes-mutation.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import assert from 'assert';
+
+export default mutation => {
+  const { name, ...patch } = mutation;
+
+  if (typeof name === 'string') {
+    assert(name.length <= 32, 'Name must be 32 characters or less');
+
+    Object.assign(patch, { name });
+  }
```
Comment (PsychoLlama) [inline src/utils/light-attributes-mutation.js]: Nitpick: `assert` could be called on `mutation.name` directly, then lines 4-5 & 8-9 wouldn't be necessary.

File: src/utils/room.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  'Front door',
+  'Garage',
+  'Terrace',
+  'Garden',
+  'Driveway',
+  'Carport',
+  'Other',
+];
+
+export const isValid = room => validRooms.includes(room);
```
Comment (PsychoLlama) [inline src/utils/room.js]: A `Set` instance would do beautifully here. `isValid` could check `validRooms.has(room)`.

Comment (PsychoLlama) [reply src/utils/room.js]: (This is part of my devious plot to raise global awareness for `Set`)

File: src/resolvers/__tests__/set-light-attributes.test.js (hunk 1 above)
Comment (PsychoLlama) [reply src/resolvers/__tests__/set-light-attributes.test.js]: Nice find! The author has some good mind thoughts. I like them 👌

Comment (mickdekkers) [reply src/resolvers/__tests__/set-light-attributes.test.js]: I'd be down with refactoring to this convention after the PR 🙂

File: src/utils/group-attributes-mutation.js (hunk 3 above)
Comment (mickdekkers) [reply src/utils/group-attributes-mutation.js]: Thanks! I always try to keep `change-case` in mind in situations like these. Super useful.

File: src/utils/light-attributes-mutation.js (hunk 4 above)
Comment (mickdekkers) [reply src/utils/light-attributes-mutation.js]: Good catch 👌

File: src/utils/room.js (hunk 5 above)
Comment (mickdekkers) [reply src/utils/room.js]: Oh yeah! Set is a thing!Consider total global awareness raised by 1 person.

File: src/utils/__tests__/group-attributes-mutation.test.js (hunk 2 above)
Comment (mickdekkers) [reply src/utils/__tests__/group-attributes-mutation.test.js]: Thanks!

**Comment by PsychoLlama** [comment]
Mighty fine 👌

**Pull request merged**